src/prefect/engine/result_handlers/s3_result_handler.py

```diff
@@ -72,7 +72,11 @@ def client(self) -> "boto3.client":
         Initializes a client if we believe we are in a new thread.
         We consider ourselves in a new thread if we haven't stored a client yet in the current context.
         """
-        if not prefect.context.get("boto3client") or not self._client:
+        if (
+            not prefect.context.get("boto3client")
+            or not hasattr(self, "_client")
+            or not self._client
```

Comment:
I think this will still raise an `AttributeError` if `self` does _not_ have the `_client` attribute.  You might want to change the last line to `or not getattr(self, "_client", None)`

Reply:
Okay will update, I thought python did short circuit evaluation in if statements